src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2257,7 +2257,8 @@ void NextPermitted(mgp_vertices_iterator &it) {
   }
 
   while (it.current_it != it.vertices.end()) {
-    if (it.graph->ctx->auth_checker->Accept(*it.graph->ctx->db_accessor, *it.current_it, it.graph->view)) {
+    if (it.graph->ctx->auth_checker->Accept(*it.graph->ctx->db_accessor, *it.current_it, it.graph->view,
+                                            memgraph::auth::FineGrainedPermission::READ)) {
```

Comment:
I guess the `WRITE` part will be checked in the function that are modifying the vertices, right?

Also one question popped up into my mind: shouldn't be `it.graph->ctx->auth_checker` checked against `nullptr`? I mgiht be wrong here.

Reply:
> I guess the `WRITE` part will be checked in the function that are modifying the vertices, right?
> 
> Also one question popped up into my mind: shouldn't be `it.graph->ctx->auth_checker` checked against `nullptr`? I mgiht be wrong here.

It is checked at the top of the method

Reply:
> I guess the `WRITE` part will be checked in the function that are modifying the vertices, right?

Correct, but still this discussion is on change that is out of scope of this PR :)

Reply:
I changed accept method in auth_checker so I needed to change it here also.